segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java

```diff
@@ -671,19 +686,25 @@ private void checkReads(HashMap<String, ByteArrayOutputStream> segmentContents,
             // This is gracefully handled by retries in AppendProcessor and/or Client, but in this case, we simply have to
             // do the retries ourselves, hoping that the callback eventually executes.
             Retry.withExpBackoff(100, 2, 10, TIMEOUT.toMillis() / 5)
-                 .retryWhen(ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException)
-                 .run(() -> {
-                     checkSegmentReads(segmentName, expectedCurrentOffset, segmentLength, store, expectedData);
-                     return null;
-                 });
+                    .retryWhen(ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException || info.get().getLength() != info.get().getStorageLength())
+                    .run(() -> {
+                        val latestInfo =  (StreamSegmentInformation) store.getStreamSegmentInfo(segmentName, TIMEOUT).join();
+                        try {
+                            checkSegmentReads(segmentName, expectedCurrentOffset, info.get().getLength(), store, expectedData);
+                        } catch (Exception ex2) {
+                            log.debug("Exception during checkReads", ex2);
+                        }
+                        info.set(latestInfo);
+                        return null;
+                    });
         }
     }
 
     private void checkSegmentReads(String segmentName, AtomicLong expectedCurrentOffset, long segmentLength, StreamSegmentStore store, byte[] expectedData) throws Exception {
         @Cleanup
         ReadResult readResult = store.read(segmentName, expectedCurrentOffset.get(), (int) (segmentLength - expectedCurrentOffset.get()), TIMEOUT).join();
         Assert.assertTrue("Empty read result for segment " + segmentName, readResult.hasNext());
-
+        boolean checkClosed = true;
```

Comment:
Isn't this equivalent to what it was before? A boolean set to `true` and then `if (checkClosed)`.

Reply:
I thought I had removed it. 
Looks I lost that change during merge.
Fixing it.

Reply:
Fixed.